### PR TITLE
fix!: consumer key prefix order to avoid complex migrations

### PR DESCRIPTION
**Pull request by smarshall-spitzbart**
## Description

The current state of main has added changes to `x/ccv/consumer/types/keys.go` s.t. a migration would be required. See [diff](https://github.com/cosmos/interchain-security/compare/v1.0.0...main) and search for `keys.go`. 

This PR restores the key prefix ordering of ICS v1.0.0, where new key prefixes are appended to the golang iota, meaning the new key prefixes use new enum values. See [diff](https://github.com/cosmos/interchain-security/compare/v1.1.0...shawn/consumer-prefix-order) and search for `keys.go` to confirm this branch does what I've claimed.

Related to https://github.com/cosmos/interchain-security/issues/939

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

* [x] Included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
* [x] Added `!` to the type prefix if API or client breaking change
* [x] Targeted the correct branch (see [PR Targeting](https://github.com/cosmos/interchain-security/blob/main/CONTRIBUTING.md#pr-targeting))
* [x] Provided a link to the relevant issue or specification
* [x] Followed the guidelines for [building SDK modules](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules)
* [x] Included the necessary unit and integration [tests](https://github.com/cosmos/interchain-security/blob/main/CONTRIBUTING.md#testing)
* [ ] Added a changelog entry to `CHANGELOG.md`
* [ ] Included comments for [documenting Go code](https://blog.golang.org/godoc)
* [ ] Updated the relevant documentation or specification
* [x] Reviewed "Files changed" and left comments if necessary 
* [ ] Confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

* [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
* [ ] confirmed `!` in the type prefix if API or client breaking change
* [ ] confirmed all author checklist items have been addressed 
* [ ] reviewed state machine logic
* [ ] reviewed API design and naming
* [ ] reviewed documentation is accurate
* [ ] reviewed tests and test coverage

### CODE REVIEW

File: x/ccv/consumer/types/keys_test.go [hunk 1]
```diff
@@ -27,18 +27,18 @@ func getAllKeyPrefixes() []byte {
...
 		ProviderChannelByteKey,
 		PendingChangesByteKey,
+		HistoricalInfoBytePrefix,
+		PacketMaturityTimeBytePrefix,
+		HeightValsetUpdateIDBytePrefix,
+		OutstandingDowntimeBytePrefix,
+		CrossChainValidatorBytePrefix,
 		PendingDataPacketsByteKey,
 		PreCCVByteKey,
 		InitialValSetByteKey,
```
Comment (smarshall-spitzbart) [inline x/ccv/consumer/types/keys_test.go:37]: Diffs in the testing file only exist to make sure the prefixes are listed in these files in the same order they're defined in `keys.go`

**Comment by mpoke** [comment]
> This PR restores the key prefix ordering of ICS v1.0.0

The issue is that ICS is currently running on the Hub, i.e., [v1.1.0-multiden](https://github.com/cosmos/gaia/blob/321d15a574def0f338ceacc5c060159ebba95edc/go.mod#L322), and on Neutron, i.e., [v1.2.0-multiden](https://github.com/neutron-org/neutron/blob/c236f1045f866c341ec26f5c409c04d201a19cde/go.mod#L14).  We need to make sure that both the Hub and Neutron can upgrade to ICS 1.3.0.

**Comment by smarshall-spitzbart** [comment]
> > This PR restores the key prefix ordering of ICS v1.0.0
> 
> The issue is that ICS is currently running on the Hub, i.e., [v1.1.0-multiden](https://github.com/cosmos/gaia/blob/321d15a574def0f338ceacc5c060159ebba95edc/go.mod#L322), and on Neutron, i.e., [v1.2.0-multiden](https://github.com/neutron-org/neutron/blob/c236f1045f866c341ec26f5c409c04d201a19cde/go.mod#L14). We need to make sure that both the Hub and Neutron can upgrade to ICS 1.3.0.

Note this PR only changes consumer keys, so we can ignore the provider in the context of the PR. (aside If you take a look at the provider `keys.go` diff between v1.1.0-multiden and main, no key migrations are required on provider). 

Re `x/ccv/consumer/types/keys.go`, this PR will allow us not to have a key prefix migration between `v1.2.0-multiden` and `v1.3.0`. See [diff](https://github.com/cosmos/interchain-security/compare/v1.2.0-multiden...shawn/consumer-prefix-order). 

TLDR: the faulty key shuffling commit is included in main, but not either of the multiden branches. Therefore this PR is still valid

**Review by mpoke — APPROVED** [review]
Approval

**Review by jtremback — APPROVED** [review]
LGTM, but is there any way to test this?

**Review by jtremback — APPROVED** [review]
LGTM, but is there any way to test this?
